evalml/data_checks/target_leakage_data_check.py

```diff
@@ -13,23 +13,25 @@
 class TargetLeakageDataCheck(DataCheck):
     """Check if any of the features are highly correlated with the target."""
 
-    def __init__(self, pct_corr_threshold=0.95):
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
         """Check if any of the features are highly correlated with the target.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, supports all target and feature types. Otherwise, only supports binary with numeric and boolean dtypes.
 
         Arguments:
             pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
```

Comment:
Just a heads up, mutual information values are normalized to be return between 0 (no mutual information) and 1 (perfect correlation). This is a bit different than pearson correlation values which are between -1 and 1. Might be worth mentioning this in the docs somewhere.

Reply:
Thanks for the info. I'll fix that.